src/dotnet-core-uninstall/Shared/Commands/ListCommandExec.cs

```diff
@@ -8,33 +8,25 @@
 using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo;
 using Microsoft.DotNet.Tools.Uninstall.Shared.Configs;
 using Microsoft.DotNet.Tools.Uninstall.Shared.Configs.Verbosity;
-using Microsoft.DotNet.Tools.Uninstall.Shared.Exceptions;
 using Microsoft.DotNet.Tools.Uninstall.Shared.Utils;
 using Microsoft.DotNet.Tools.Uninstall.Shared.VSVersioning;
-using Microsoft.DotNet.Tools.Uninstall.Windows;
 
 namespace Microsoft.DotNet.Tools.Uninstall.Shared.Commands
 {
     internal static class ListCommandExec
     {
-        public static void Execute()
+        private static IConsole SysConsole;
+        private static Region Region;
+        private static string[] Args;
+
+        public static void Execute(IBundleCollector bundleCollector, IConsole console = null, Region region = null, string[] args = null)
         {
-            if (RuntimeInfo.RunningOnWindows)
-            {
-                Execute(
-                    RegistryQuery.GetAllInstalledBundles(),
-                    Windows.SupportedBundleTypeConfigs.SupportedBundleTypes);
-            }
-            else if (RuntimeInfo.RunningOnOSX)
-            {
-                Execute(
-                    FileSystemExplorer.GetAllInstalledBundles(),
-                    MacOs.SupportedBundleTypeConfigs.SupportedBundleTypes);
-            }
-            else
-            {
-                throw new OperatingSystemNotSupportedException();
-            }
+            SysConsole = console == null? new SystemConsole() : console;
```

Comment:
suggestion: Looks like there are multiple bugs. It is better to have separate commits with tests per bug, so the viewer (and reader after it is checked in) know which bug corresponding to which test.

If it is hard to separate. You can just add some pull request comments to point them out. Like "this line is for bug A" "this line is the test for bug A"

Reply:
And here i really don't know if it is a refactoring about SysConsole init or it is a bug

Reply:
Same for refactoring and feature mix. Ideally refactoring commit should be separated with feature/bug fix. Unless it is obvious to separate them out like they are in different files.

If it hard to separate, it's better to add some comment in PR.